nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/consumer/watermarker/JpegWatermarker.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (C) Copyright 2018 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.importer.stream.consumer.watermarker;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.apache.commons.imaging.ImageReadException;
+import org.apache.commons.imaging.ImageWriteException;
+import org.apache.commons.imaging.Imaging;
+import org.apache.commons.imaging.common.ImageMetadata;
+import org.apache.commons.imaging.formats.jpeg.JpegImageMetadata;
+import org.apache.commons.imaging.formats.jpeg.exif.ExifRewriter;
+import org.apache.commons.imaging.formats.tiff.TiffImageMetadata;
+import org.apache.commons.imaging.formats.tiff.constants.ExifTagConstants;
+import org.apache.commons.imaging.formats.tiff.write.TiffOutputDirectory;
+import org.apache.commons.imaging.formats.tiff.write.TiffOutputSet;
+import org.apache.commons.io.FileUtils;
+
+/**
+ * @since 10.2
+ */
+public class JpegWatermarker extends AbstractWatermarker {
+
+    @Override
+    public Path addWatermark(Path inputFile, Path outputDir, String watermark) {
+        File jpegImageFile = inputFile.toFile();
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
```

Comment:
You should not use an in-memory output stream to store the image, it may be huge. Use a temporary file instead.

Reply:
yes or no because the goal is to watermark as fast as possible, so we don't want to create more disk IO than needed even if it cost memory.
This code is only for generating tests data for benchmarking it is not intended to be used in production.